### ISSUE TITLE
When using sidecar with nREPL, :preloads flag can't find namespace unless specify cljs source path in clj :source-paths

### ISSUE THREAD

**Issue by visibletrap** [bhauman/lein-figwheel#566]
Normally, I separate Clojure and Clojurescript sources to `src/clj` and `src/cljs`. And I set `src/clj` to leiningen's root-level `source-paths` and specify `src/cljs` to source-paths in cljs compiler option. I have been using all these three approach to start figwheel, lein-fighweel, clojure.main repl + sidecar and nrepl + sidecar, everything has been working great.

When I add `:preloads` flag to cljs compiler config and starts with nrepl. Figwheel shows this warning
```
WARNING: preload namespace my.namespace does not exist
```
I also can't load that namespace in the figwheel repl. This issue doesn't appear when using figwheel via lein-figwheel and clojure.main repl.

I have found a workaround for this. Adding `src/cljs` to root-level source-paths fixes the issue. There's no more warning and the namespace is loaded correctly in figwheel repl.

**Comment by bhauman** [comment]
When you config outside of leinigen you need to quote your symbols.

Leinigen auto-quotes all raw symbols in the project.clj

**Comment by visibletrap** [comment]
Thanks for a quick response, Bruce.

I have tried both using config from leiningen (`(start-figwheel!)` without any argument) and setting an argument to `(start-figwheel! config)` with `:preloads '[my.namespace]`. In both cases, there's a warning shows up.

I have created [a project to demonstrate this](https://github.com/visibletrap/preload-can-not-find-ns). It's generated by leiningen's figwheel template with [a few modifications](https://github.com/visibletrap/preload-can-not-find-ns/commit/0eddbc2db97eda51b84f5b0874a364f00ae2c108).

If you have time, can you clone it and try these.
 * `lein figwheel`: no warning
 * Run `lein repl`, then run `(fig-start)`: warning shows up
 * Run `lein repl`, then run `(fig-start-with-direct-args)`: warning shows up

Then try replacing `:source-paths ["src/clj" "dev"]` with `:source-paths ["src/clj" "dev" "src/cljs"]`

 * Run `lein repl`, then run `(fig-start)`: no warning
 * Run `lein repl`, then run `(fig-start-with-direct-args)`: no warning

Thank you so much

**Comment by bhauman** [comment]
This is actually the normal behavior, lein figwheel adds the cljs
:source-paths to the lein root :source-paths in the figwheel plugin.  When
you use figwheel without the plugin you need to add these paths to the
:source-paths as you have done in your example.

There is no good way to fix this behavior, as you have to set it up when
you start the JVM process.

**Comment by visibletrap** [comment]
Got it. Thank you so much for the explanation. I'm good with adding path myself.

For people who later landed to this issue, I deleted my example project. I captured changes into images here

<img width="1244" alt="screen shot 2017-07-04 at 11 23 59 am" src="https://user-images.githubusercontent.com/625523/27815060-8dfc8262-60ab-11e7-870a-585f21f08c2e.png">
<img width="1241" alt="screen shot 2017-07-04 at 11 24 12 am" src="https://user-images.githubusercontent.com/625523/27815063-91beff38-60ab-11e7-8a62-c0a762449275.png">
<img width="1239" alt="screen shot 2017-07-04 at 11 24 19 am" src="https://user-images.githubusercontent.com/625523/27815066-9551f56a-60ab-11e7-8ec8-cf095743ef6e.png">
